File: Tanker/Sources/TKREncryptionOptions.m
#import <Tanker/TKREncryptionOptions.h>

@implementation TKREncryptionOptions

+ (instancetype)options
{
  TKREncryptionOptions* opts = [[self alloc] init];
  opts.shareWithUsers = @[];
  opts.shareWithGroups = @[];
  opts.shareWithSelf = true;
  opts.paddingStep = [TKRPadding automatic];
  return opts;
}

@end

File: Tanker/Sources/TKREncryptionOptions.swift
@objc(TKREncryptionOptions)
public class EncryptionOptions: NSObject {
  @objc
  public var shareWithUsers: Array<String>;
  @objc
  public var shareWithGroups: Array<String>;
  @objc
  public var shareWithSelf: Bool;
  @objc
  public var paddingStep: Padding;
  
  @objc
  public override init() {
    self.shareWithUsers = [];
    self.shareWithGroups = [];
    self.shareWithSelf = true;
    self.paddingStep = Padding.automatic()!;
  }
}
